spec/updateDispatchDateSpec.js

```diff
@@ -0,0 +1,124 @@
+"use strict";
+
+const updateDispatchDate = require('../lib/actions/updateDispatchDate.js').process;
+const nock = require('nock');
+
+nock.disableNetConnect();
+
+console.log = () => {};
+
+// fixtures
+const input = require('./fixtures/updateDispatchDate/palletInput.json');
+const nonPalletInput = require('./fixtures/updateDispatchDate/nonPalletInput.json');
+
+describe("Updating a pallet order", () => {
+  beforeEach(() => {
+    this.msg = {
+      body: input
+    }
+
+    this.cfg = {
+      host:   "https://vendors-staging.herokuapp.com",
+      apiKey: "abc123"
+    }
+  });
+
+  describe("Successful update", () => {
+    beforeEach((done) => {
+      this.self = {
+        emit(action) { if (action == 'end') done(); }
+      };
+      spyOn(this.self, 'emit').and.callThrough();
+
+      this.updateOrderRequest = nock('https://vendors-staging.herokuapp.com')
+        .patch('/api/v1/orders/123', {
+          "data": {
+            "id": "123",
+            "type": "orders",
+            "attributes": {
+              "dispatch-date": "2016-11-05T00:00:00+00:00"
+            }
+          }
+        });
+
+      this.updateOrderRequest = this.updateOrderRequest.reply(200, {});
+
+      updateDispatchDate.call(this.self, this.msg, this.cfg);
+    });
+
+    it("Sends a request to the VMS", () => {
+      expect(this.updateOrderRequest.isDone()).toEqual(true);
+    });
+
+    it("Emits the data", () => {
```

Comment:
What kind of data does it emit? It might be worth stating that it emits the original message and then checking for that in the spec below.